File: scala_2.9.1/src/main/scala/thirtytwo/degrees/halfpipe/scalaexample/web/MvcConfig.scala
package thirtytwo.degrees.halfpipe.scalaexample.web

import org.springframework.web.servlet.config.annotation.{ResourceHandlerRegistry, WebMvcConfigurerAdapter, EnableWebMvc}
import org.springframework.context.annotation.{ComponentScan, Configuration}
import thirtytwo.degrees.halfpipe.web.admin.AdminControllers

/**
 * User: gibbsb
 * Date: 9/22/12
 * Time: 4:11 PM
 */
@EnableWebMvc
@Configuration
@ComponentScan(basePackageClasses = Array(classOf[MvcConfig], classOf[AdminControllers]))
class MvcConfig extends WebMvcConfigurerAdapter {

  override def addResourceHandlers(registry: ResourceHandlerRegistry) {
    registry.addResourceHandler("/**").addResourceLocations("classpath:/static/")
  }
}

File: src/main/java/thirtytwo/degrees/halfpipe/example/web/MvcConfig.java
package thirtytwo.degrees.halfpipe.example.web;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.*;
import thirtytwo.degrees.halfpipe.web.admin.AdminControllers;

/**
 * User: spencer
 * Date: 9/21/12
 * Time: 4:18 PM
 * <p/>
 * http://static.springsource.org/spring/docs/3.1.x/spring-framework-reference/html/mvc.html#mvc-config-enable
 */
@EnableWebMvc
@Configuration
@ComponentScan(basePackageClasses = {MvcConfig.class, AdminControllers.class})
public class MvcConfig extends WebMvcConfigurerAdapter {

    @Override
    public void configureDefaultServletHandling(DefaultServletHandlerConfigurer configurer) {
        configurer.enable();
    }

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        registry.addResourceHandler("/**")
                .addResourceLocations("classpath:/static/");
    }
}
